crates/stackable-operator/src/crd/git_sync/mod.rs

```diff
@@ -63,5 +64,13 @@ pub mod versioned {
         /// [example]: DOCS_BASE_URL_PLACEHOLDER/airflow/usage-guide/mounting-dags#_example
         #[serde(default)]
         pub git_sync_conf: BTreeMap<String, String>,
+
+        /// The name of the Secret used for SSH access to the repository.
+        ///
+        /// The referenced Secret must include two fields: `key` and `knownHosts`.
+        /// This cannot be provided if `credentials_secret` is also provided.
```

Comment:
note: Ideally we want this to be a single field instead which is an enum with two variants. This completely prevents the potential misuse by users.

I know this is a breaking change, but we should consider it going forward.